common/surroundings_loader.rs

```diff
@@ -7,43 +7,10 @@ use std::cmp::max;
 use std::collections::VecDeque;
 use stopwatch;
 
-use cube_shell::cube_diff;
+use cube_shell::{cube_diff, cube_shell};
 
-mod surroundings_iter {
-  use cgmath::{Point3};
-  use std;
-
-  use cube_shell::cube_shell;
-
-  pub struct CubeShellClosure {
-    center: Point3<i32>,
-  }
-
-  impl FnOnce<(i32,)> for CubeShellClosure {
-    type Output = std::vec::IntoIter<Point3<i32>>;
-    extern "rust-call" fn call_once(self, radius: (i32,)) -> Self::Output {
-      cube_shell(&self.center, radius.0).into_iter()
-    }
-  }
-
-  impl FnMut<(i32,)> for CubeShellClosure {
-    extern "rust-call" fn call_mut(&mut self, radius: (i32,)) -> Self::Output {
-      cube_shell(&self.center, radius.0).into_iter()
-    }
-  }
-
-  pub type T =
-    std::iter::FlatMap<
-      std::ops::Range<i32>,
-      std::vec::IntoIter<Point3<i32>>,
-      CubeShellClosure,
-    >;
-
-  pub fn new(center: &Point3<i32>, max_distance: i32) -> T {
-    (0 .. max_distance).flat_map(CubeShellClosure {
-      center: *center,
-    })
-  }
+fn surroundings_iter(center: Point3<i32>, max_distance: i32) -> Box<Iterator<Item=Point3<i32>> + Send> {
+  Box::new((0 .. max_distance).flat_map(move |radius| cube_shell(&center, radius)))
```

Comment:
I'm sad about this but per [this thread](https://users.rust-lang.org/t/how-do-you-store-the-result-of-an-impl-trait-return-type/14808/7) we still can't use existentials to eliminate the `Box`.